File: codex_poting/java/src/main/java/com/example/demo/mybatis/mapper/BookStockCustomMapper.java
package com.example.demo.mybatis.mapper;

import com.example.demo.mybatis.generator.entity.BookStockEntity;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

@Mapper
public interface BookStockCustomMapper {
    BookStockEntity selectByStoreIdAndBookIdWithWriteLock(
        @Param("bookStockStoreId") Long bookStockStoreId,
        @Param("bookStockBookId") Long bookStockBookId
    );
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/mybatis/mapper/BookStockCustomMapper.kt
package com.example.demo.mybatis.mapper

import com.example.demo.mybatis.generator.entity.BookStockEntity
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface BookStockCustomMapper {
    fun selectByStoreIdAndBookIdWithWriteLock(
        @Param("bookStockStoreId") bookStockStoreId: Long?,
        @Param("bookStockBookId") bookStockBookId: Long?
    ): BookStockEntity?
}
